src/Native/System.Native/pal_networkchange.cpp

```diff
@@ -52,7 +51,11 @@ extern "C" NetworkChangeKind SystemNative_ReadSingleEvent(int32_t sock)
     char buffer[4096];
     iovec iov = {buffer, sizeof(buffer)};
     sockaddr_nl sanl;
+#if MSGHDR_REQUIRES_PADDING
+    msghdr msg = {reinterpret_cast<void*>(&sanl), sizeof(sockaddr_nl), &iov, 1, 0, NULL, 0, 0, 0};
+#else
     msghdr msg = {reinterpret_cast<void*>(&sanl), sizeof(sockaddr_nl), &iov, 1, NULL, 0, 0};
+#endif
```

Comment:
The only tests we have for the NetworkChange events are very basic. They just verify that they are able to subscribe and then unsubscribe to the event. I did more manual testing of NetworkChange when I was working on the implementation but it's possible there is an issue hidden here.